app/src/main/java/com/siddharth/practiceapp/adapter/HomeRvAdapter.kt

```diff
@@ -1,97 +1,76 @@
 package com.siddharth.practiceapp.adapter
 
-import android.os.Build
-import android.util.Log
 import android.view.LayoutInflater
-import android.view.View
 import android.view.ViewGroup
-import android.widget.ImageView
-import android.widget.TextView
-import androidx.annotation.RequiresApi
 import androidx.core.view.isVisible
-import androidx.recyclerview.widget.RecyclerView
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.ListAdapter
+import androidx.recyclerview.widget.RecyclerView.ViewHolder
+import androidx.viewbinding.ViewBinding
 import com.bumptech.glide.Glide
-import com.siddharth.practiceapp.R
 import com.siddharth.practiceapp.data.entities.HomeData
+import com.siddharth.practiceapp.databinding.ItemNewsHomeBinding
+import com.siddharth.practiceapp.databinding.ItemRemainderHomeBinding
 import com.siddharth.practiceapp.util.slideUp
 
-class HomeRvAdapter : RecyclerView.Adapter<RecyclerView.ViewHolder>() {
+class HomeRvAdapter : ListAdapter<HomeData, HomeRvAdapter.MyAbstractViewHolder>(COMPARATOR) {
 
     companion object {
         @JvmStatic
         val newsType = 1
 
         @JvmStatic
         val reminderType = 2
-    }
 
-    private var positionForSpeedItem = -1
-    private var shouldShouldSpeedItem = false
-    val dataList = ArrayList<HomeData>()
+        private val COMPARATOR = object : DiffUtil.ItemCallback<HomeData>(){
+            override fun areContentsTheSame(oldItem: HomeData, newItem: HomeData): Boolean {
+                return oldItem.id == newItem.id
+            }
 
-    class NewsHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
-        init {
-            Log.d("NewsHolder", "created")
+            override fun areItemsTheSame(oldItem: HomeData, newItem: HomeData): Boolean {
+                return oldItem == newItem
+            }
         }
+    }
 
-        private val newsHeadline: TextView = itemView.findViewById(R.id.tv_newsHeadline)
-        private val newsDescription: TextView = itemView.findViewById(R.id.tv_newsDesc)
-        // private val newsView = itemView as NewsItemView
+    private var positionForSpeedItem = -1
+    private var shouldShouldSpeedItem = false
 
-        @RequiresApi(Build.VERSION_CODES.O)
-        fun setData(data: HomeData, holder: NewsHolder, position: Int) {
-            holder.newsHeadline.text = data.title
-            holder.newsDescription.text = data.description
-            // newsView.populate(data)
+    class NewsHolder(private val binding: ItemNewsHomeBinding) : MyAbstractViewHolder(binding) {
```

Comment:
you can make this and other viewholders private also, they are not required to be reusable.